runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/BatchDataflowWorker.java

```diff
@@ -214,22 +215,34 @@ protected BatchDataflowWorker(
     // TODO: this conditional -> two implementations of common interface, or
     // param/injection
     if (DataflowRunner.hasExperiment(options, "beam_fn_api")) {
-      Function<MutableNetwork<Node, Edge>, Node> sdkFusedStage =
-          pipeline == null
-              ? RegisterNodeFunction.withoutPipeline(
-                  IdGenerator::generate, sdkHarnessRegistry.beamFnStateApiServiceDescriptor())
-              : RegisterNodeFunction.forPipeline(
-                  pipeline,
-                  IdGenerator::generate,
-                  sdkHarnessRegistry.beamFnStateApiServiceDescriptor());
+      Function<MutableNetwork<Node, Edge>, MutableNetwork<Node, Edge>> transformToRunnerNetwork;
+      Function<MutableNetwork<Node, Edge>, Node> sdkFusedStage;
       Function<MutableNetwork<Node, Edge>, MutableNetwork<Node, Edge>> lengthPrefixUnknownCoders =
           LengthPrefixUnknownCoders::forSdkNetwork;
-      Function<MutableNetwork<Node, Edge>, MutableNetwork<Node, Edge>> transformToRunnerNetwork =
-          new CreateRegisterFnOperationFunction(
-              IdGenerator::generate,
-              this::createPortNode,
-              lengthPrefixUnknownCoders.andThen(sdkFusedStage));
-
+      if (DataflowRunner.hasExperiment(options, "use_executable_stage_bundle_execution")) {
+        sdkFusedStage = new CreateExecutableStageNodeFunction(pipeline, IdGenerator::generate);
+        transformToRunnerNetwork =
+            new CreateRegisterFnOperationFunction(
```

Comment:
Why do we need the CreateRegisterFnOperationFunction still, isn't registration handled by the RemoteBundle returned by the RemoteStageBundleFactory?

Reply:
discussed offline. CreateRegisterFnOperationFunction creates the sdk fused graph which is also needed in constructing ExecutableStage.